extensions-core/kubernetes-extensions/src/main/java/org/apache/druid/k8s/discovery/DefaultK8sLeaderElectorFactory.java

```diff
@@ -56,7 +62,22 @@ public K8sLeaderElector create(String candidateId, String namespace, String lock
             Duration.ofMillis(discoveryConfig.getRenewDeadline().getMillis()),
             Duration.ofMillis(discoveryConfig.getRetryPeriod().getMillis())
         );
-    LeaderElector leaderElector = new LeaderElector(leaderElectionConfig);
+    LeaderElector leaderElector = new LeaderElector(leaderElectionConfig, (t) -> {
+      // If any errors occur reading the configmap overwrite it and force a fresh election
+      LOGGER.warn(t, "Failed to get elect leader for [%s], overwriting invalid ConfigMaps", lockResourceName);
+      Date now = new Date();
+      boolean result = lock.update(new LeaderElectionRecord(
```

Comment:
will this potentially fail with the same date serialization issue that we encountered? this seems like a once retry. or it will keep retrying? 🤔

Reply:
It will retry every time the leader election routine runs